### DISCUSSION TITLE
Model and Screen specific Background Image / Screen filling Widgets?

### DISCUSSION THREAD

**Discussion by Durahl** [EdgeTX/edgetx#1864] [Ideas]
Greetings!

I'm still on **OpenTX** until all the parts for converting my X12S to a touch model have arrived - Just to _explain_ my question.

Does **EdgeTX** have the possibility of displaying a Screen **AND** Model specific Image that can be used as a Quickstart Guide for the Model by displaying the Radios Settings? Kinda like how Games have a QuickStart Guide Controller Page showing you what Button A or the R1 does? 🤔

I have a bunch of [Tamiya RC Tanks](https://imgur.com/a/qyme4mt) which generally have the same layout but two of them are modified with additional hardware adding further complexity to the already extensive Switch Layout ( not a big Fan of the Tamiya "Konami" thing for accessing functions beyond 4CH limitation so [everything has been reworked to using Switches](https://youtu.be/VNP6Dstz6wc) instead of Sticks and Trimmers ) and without frequent use I keep forgetting this stuff 🤣

**OpenTX** does neither have a reasonably sized Widget to house such an QSG Image or the ability to display Screen specific ( or even just Model specific - though not preferred - Background Images ) which would allow me to setup Screen _I_ for Telemetry, Screen _II_ for Channels with both using my regular "beauty" Background Image while Screen _III_ would be displaying the Model Specific Radio Control QSG as either a _different_ Background image or via a screen filling Widget?

Cheers! 😁

**Comment by pfeerick** [comment]
You can have model notes to remind you what all your switches are for. But the nearest I know to of a widget that might be able to help is the [Switch2](https://www.rcgroups.com/forums/showpost.php?p=47009445&postcount=3793) widget, which can show icons for all the various switches (and their current state). 

Model level background and theme options are planned and is being tracked here https://github.com/EdgeTX/edgetx/issues/1681

**Reply by lshems** [reply]
There is no reason you couldn't write a widget to show a standard named image.

Actually, that is one of the easiest widgets as it doesn't need workflow or interactions .

You can check my blog on rcgroups for moe info, and there is a lua thread there that helps as well.

**Comment by Durahl** [comment]
**@pfeerick** Yes, I know about the Notes Option but I'm kinda a visual person who likes to look at cutesy pictures 😏
But I'll be having a look at that suggested Widget! How much I'll be liking it will mostly depend on how well the Iconography works 🤔

**@lshems** Yeeaaa... The best I think I can do is _modify_ an existing Script to suite my needs but even that I'd be describing as _highly_ optimistic.
_Writing_ my own script from zero knowledge of the LUA language ( /cough or coding in a broad sense /cough ) is probably best described as: _"Outside the scope of this video™"_ - **ToT** 😁

But just for having a laugh on my account... [Which Topic](https://www.rcgroups.com/forums/member.php?u=577751) were you talking about? 🤨

**Comment by lshems** [comment]
https://www.rcgroups.com/forums/showthread.php?2180477-LUA-scripting-Technical-discussion

**Comment by lshems** [comment]
Perhaps I'll make a widget for that.

**Reply by Durahl** [reply]
![IMG](https://i.imgur.com/Ls8yhxE.gif)

**Comment by lshems** [comment]
I need a little help for this widget. I wanted to just recreate the modelBmp widget in LUA, as it exist in OpenTX. I just am not sure it already exists in EdgeTX.

If so, what is this widget's name and where do i put the bmp?

I was thinking of adding an option to the widget to select a number. 0 will simply be modelname1.bmp or modelname.bmp if model1.bmp does not exist. Number 2 will load modelname2.bmp.

I will just scale the bmp to the widget size.

**Comment by lshems** [comment]
OK, found it. Models. It uses the model name if no bitmap selected, and the model bitmap if selected. It scales properly, and you can set some color and fill options.

SO if I would rebuild that widget, but would add an extra number parameter, you could select any picture with the same name as the model where that number is added to the file name selected on the model.

would that fit the requirements?

**Reply by Durahl** [reply]
I don't quite understand the "same name as the model" part 🤔

In [this Video](https://youtu.be/VNP6Dstz6wc) there's two Pictures being used:
- Background ( the character on the left side of the screen - Displayed regardless of active model )
- Leo2A7 ( model specific picture displayed on both the characters chest and the model selector )

Are you implying to just be doing something like a Leo2A7_1?

As for additional options - Hard to say... I guess being able to choose a background color in case the image is using transparency would be a requirement for some users? 🤨
